src/Hooks.php

```diff
@@ -6,8 +6,6 @@
 
 class Hooks {
 
-	use Singleton;
-
 	private $API;
 
 	public function __construct() {
```

Comment:
Lets use dependency injection by changing the constructor, this object requires an `API` object to do it's work, so lets make it ask for one when it gets created:

```php
public function __construct( API $api ) {
    $this->api = $api;
}
```

Now we don't need to check if api is set, if it isn't set the constructor will throw an error thanks to the type hinting